bands/site_sections/bands.py

```diff
@@ -9,128 +9,159 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, band_id, id=None, message='', **params):
+        band = session.band(band_id)
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
                 message = 'You must tell us how many people are in your band'
-            elif not band.poc_phone:
+            elif not band_info.poc_phone:
                 message = 'You must enter an on-site point-of-contact cellphone number'
-            elif not band.arrival_time:
+            elif not band_info.arrival_time:
                 message = 'You must enter your expected arrival time'
-            elif band.bringing_vehicle and not band.vehicle_info:
+            elif band_info.bringing_vehicle and not band_info.vehicle_info:
                 message = 'You must provide your vehicle information'
             else:
+                band.info = band_info
+                session.add(band)
```

Comment:
You don't need to say ``session.add(band)`` here or in any of the other forms on the checklist, because when we say ``session.band(band_id)`` we're by definition getting something that's already in the database, so it doesn't need to be added, and changes to the object will be automatically saved on commit (which itself happens automatically on redirect).

The second ``session.add(band_info)`` is of course still necessary, since that may or may not exist yet.